airflow/api_fastapi/core_api/routes/public/xcom.py

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import copy
+
+from fastapi import Depends, HTTPException
+from sqlalchemy import and_, select
+from sqlalchemy.orm import Session
+from typing_extensions import Annotated
+
+from airflow.api_fastapi.common.db.common import get_session
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.serializers.xcom import (
+    XComResponseNative,
+    XComResponseString,
+)
+from airflow.models import DagRun as DR, XCom
+from airflow.settings import conf
+
+xcom_router = AirflowRouter(
+    tags=["XCom"], prefix="/dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}/xcomEntries"
+)
+
+
+@xcom_router.get(
+    "/{xcom_key}",
+    responses=create_openapi_http_exception_doc([400, 401, 403, 404]),
+)
+async def get_xcom_entry(
```

Comment:
```suggestion
def get_xcom_entry(
```

Convention has just changed, route should be declared `sync`. Sorry for the annoyance.